src/Database/Type/DateTimeType.php

```diff
@@ -165,7 +169,7 @@ public function marshal($value)
                 $date = new $class($value);
                 $compare = true;
             }
-            if ($compare && $date && $date->format($this->_format) !== $value) {
+            if ($compare && $date && !$this->_compare($date, $value)) {
```

Comment:
Do we really need to check whether the date string can be restored here? I think this kind of check should be done by validation. If we could remove this check, people could use `<input type="datetime-local">` fields, that uses another date string formats, such as `Y-m-d\TH:i` or  `Y-m-d\TH:i:s`.
Also, I think we have to change `Validation::datetime()` too. At the moment, it doesn't support for ISO datetime strings.